### PR TITLE
Fix #142

### LINKED ISSUE #142: Remove non-SSL communication with external hosts
We will not be supporting non-SSL communication with external hosts, like our API endpoints or the WP.org API endpoints.  We'll put a check for `! wp_http_supports( array( 'ssl' ) )` into the migration plugin and deny the migration.

So all of these `wp_http_supports( array( 'ssl' ) )` calls need to go, and we can just build the corresponding URLs with `https:` right away:

```
$ ack -C3 wp_http_supports.*ssl
tests/phpunit/tests/http/base.php:383-	 * @see https://core.trac.wordpress.org/ticket/25007
tests/phpunit/tests/http/base.php:384-	 */
tests/phpunit/tests/http/base.php:385-	function test_ssl() {
tests/phpunit/tests/http/base.php:386:		if ( ! wp_http_supports( array( 'ssl' ) ) )
tests/phpunit/tests/http/base.php:387-			$this->fail( 'This installation of PHP does not support SSL' );
tests/phpunit/tests/http/base.php:388-
tests/phpunit/tests/http/base.php:389-		$res = wp_remote_get( 'https://wordpress.org/' );
src/wp-includes/update.php:128-		$post_body = array_merge( $post_body, $extra_stats );
src/wp-includes/update.php:129-
src/wp-includes/update.php:130-	$url = $http_url = 'https://api-v1.classicpress.net/core/version-check/1.0/?' . http_build_query( $query, null, '&' );
src/wp-includes/update.php:131:	if ( $ssl = wp_http_supports( array( 'ssl' ) ) )
src/wp-includes/update.php:132-		$url = set_url_scheme( $url, 'https' );
src/wp-includes/update.php:133-
src/wp-includes/update.php:134-	$doing_cron = wp_doing_cron();
--
src/wp-includes/update.php:332-	}
src/wp-includes/update.php:333-
src/wp-includes/update.php:334-	$url = $http_url = 'http://api.wordpress.org/plugins/update-check/1.1/';
src/wp-includes/update.php:335:	if ( $ssl = wp_http_supports( array( 'ssl' ) ) )
src/wp-includes/update.php:336-		$url = set_url_scheme( $url, 'https' );
src/wp-includes/update.php:337-
src/wp-includes/update.php:338-	$raw_response = wp_remote_post( $url, $options );
--
src/wp-includes/update.php:514-	}
src/wp-includes/update.php:515-
src/wp-includes/update.php:516-	$url = $http_url = 'http://api.wordpress.org/themes/update-check/1.1/';
src/wp-includes/update.php:517:	if ( $ssl = wp_http_supports( array( 'ssl' ) ) )
src/wp-includes/update.php:518-		$url = set_url_scheme( $url, 'https' );
src/wp-includes/update.php:519-
src/wp-includes/update.php:520-	$raw_response = wp_remote_post( $url, $options );
src/wp-admin/includes/class-wp-community-events.php:99-		$request_args = $this->get_request_args( $location_search, $timezone );
src/wp-admin/includes/class-wp-community-events.php:100-		$request_args['user-agent'] = 'ClassicPress/' . $wp_version . '; ' . home_url( '/' );
src/wp-admin/includes/class-wp-community-events.php:101-
src/wp-admin/includes/class-wp-community-events.php:102:		if ( wp_http_supports( array( 'ssl' ) ) ) {
src/wp-admin/includes/class-wp-community-events.php:103-			$api_url = set_url_scheme( $api_url, 'https' );
src/wp-admin/includes/class-wp-community-events.php:104-		}
src/wp-admin/includes/class-wp-community-events.php:105-
src/wp-admin/includes/dashboard.php:1518-			'user-agent' => 'ClassicPress/' . $wp_version . '; ' . home_url( '/' )
src/wp-admin/includes/dashboard.php:1519-		);
src/wp-admin/includes/dashboard.php:1520-
src/wp-admin/includes/dashboard.php:1521:		if ( wp_http_supports( array( 'ssl' ) ) ) {
src/wp-admin/includes/dashboard.php:1522-			$url = set_url_scheme( $url, 'https' );
src/wp-admin/includes/dashboard.php:1523-		}
src/wp-admin/includes/dashboard.php:1524-
src/wp-admin/includes/theme.php:445-		include( ABSPATH . WPINC . '/version.php' );
src/wp-admin/includes/theme.php:446-
src/wp-admin/includes/theme.php:447-		$url = $http_url = 'http://api.wordpress.org/themes/info/1.0/';
src/wp-admin/includes/theme.php:448:		if ( $ssl = wp_http_supports( array( 'ssl' ) ) )
src/wp-admin/includes/theme.php:449-			$url = set_url_scheme( $url, 'https' );
src/wp-admin/includes/theme.php:450-
src/wp-admin/includes/theme.php:451-		$http_args = array(
src/wp-admin/includes/update.php:103-function get_core_checksums( $version, $locale ) {
src/wp-admin/includes/update.php:104-	$url = $http_url = 'http://api.wordpress.org/core/checksums/1.0/?' . http_build_query( compact( 'version', 'locale' ), null, '&' );
src/wp-admin/includes/update.php:105-
src/wp-admin/includes/update.php:106:	if ( $ssl = wp_http_supports( array( 'ssl' ) ) )
src/wp-admin/includes/update.php:107-		$url = set_url_scheme( $url, 'https' );
src/wp-admin/includes/update.php:108-
src/wp-admin/includes/update.php:109-	$options = array(
src/wp-admin/update-core.php:164-		echo '<h2>';
src/wp-admin/update-core.php:165-		_e('You have the latest version of ClassicPress.');
src/wp-admin/update-core.php:166-
src/wp-admin/update-core.php:167:		if ( wp_http_supports( array( 'ssl' ) ) ) {
src/wp-admin/update-core.php:168-			require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
src/wp-admin/update-core.php:169-			$upgrader = new WP_Automatic_Updater;
src/wp-admin/update-core.php:170-			$future_minor_update = (object) array(
--
src/wp-admin/update-core.php:191-	if ( isset( $updates[0] ) && $updates[0]->response == 'development' ) {
src/wp-admin/update-core.php:192-		require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
src/wp-admin/update-core.php:193-		$upgrader = new WP_Automatic_Updater;
src/wp-admin/update-core.php:194:		if ( wp_http_supports( 'ssl' ) && $upgrader->should_update( 'core', $updates[0], ABSPATH ) ) {
src/wp-admin/update-core.php:195-			echo '<div class="updated inline"><p>';
src/wp-admin/update-core.php:196-			echo '<strong>' . __( 'BETA TESTERS:' ) . '</strong> ' . __( 'This site is set up to install updates of future beta versions automatically.' );
src/wp-admin/update-core.php:197-			echo '</p></div>';
src/wp-admin/includes/translation-install.php:36-
src/wp-admin/includes/translation-install.php:37-	if ( false === $res ) {
src/wp-admin/includes/translation-install.php:38-		$url = $http_url = 'http://api.wordpress.org/translations/' . $type . '/1.0/';
src/wp-admin/includes/translation-install.php:39:		if ( $ssl = wp_http_supports( array( 'ssl' ) ) ) {
src/wp-admin/includes/translation-install.php:40-			$url = set_url_scheme( $url, 'https' );
src/wp-admin/includes/translation-install.php:41-		}
src/wp-admin/includes/translation-install.php:42-
src/wp-admin/includes/plugin-install.php:145-		include( ABSPATH . WPINC . '/version.php' );
src/wp-admin/includes/plugin-install.php:146-
src/wp-admin/includes/plugin-install.php:147-		$url = $http_url = 'http://api.wordpress.org/plugins/info/1.0/';
src/wp-admin/includes/plugin-install.php:148:		if ( $ssl = wp_http_supports( array( 'ssl' ) ) )
src/wp-admin/includes/plugin-install.php:149-			$url = set_url_scheme( $url, 'https' );
src/wp-admin/includes/plugin-install.php:150-
src/wp-admin/includes/plugin-install.php:151-		$http_args = array(
src/wp-admin/includes/import.php:136-		), 'http://api.wordpress.org/core/importers/1.1/' );
src/wp-admin/includes/import.php:137-		$options = array( 'user-agent' => 'ClassicPress/' . $wp_version . '; ' . home_url( '/' ) );
src/wp-admin/includes/import.php:138-
src/wp-admin/includes/import.php:139:		if ( wp_http_supports( array( 'ssl' ) ) ) {
src/wp-admin/includes/import.php:140-			$url = set_url_scheme( $url, 'https' );
src/wp-admin/includes/import.php:141-		}
src/wp-admin/includes/import.php:142-
```

### PR DESCRIPTION
**Pull request by invisnet**

### CODE REVIEW

File: src/wp-includes/http.php [hunk 1]
```diff
@@ -365,25 +366,7 @@ function wp_remote_retrieve_cookie_value( $response, $name ) {
  * @return bool
  */
 function wp_http_supports( $capabilities = array(), $url = null ) {
```
Comment (nylen) [inline src/wp-includes/http.php]: Since this is `@deprecated` now, we should call `_deprecated_function( __FUNCTION__, '1.0.0-beta1' );` here too.

**Comment by nylen** [comment]
LGTM apart from the missing `_deprecated_function` call.  If that's intentional let me know, otherwise let's add that before merging.

File: src/wp-includes/http.php (hunk 1 above)
Comment (invisnet) [reply src/wp-includes/http.php]: Yes and no - if we do that now we generate a lot of noise in plugins that are unlikely to add suitable guards.

Comment (nylen) [reply src/wp-includes/http.php]: OK, let's go with this for now then.

**Pull request merged**